plugins/cake/cake_

```diff
@@ -0,0 +1,202 @@
+#!/bin/bash
+# -*- sh -*-
+
+: << =cut
+
+=pod
+
+=encoding UTF-8
+
+=head1 NAME
+
+cake_ - Plugin to monitor cake's backlog, dropped, overlimits and requeues
+
+=head1 CONFIGURATION
+
+None needed.
+
+=head1 INTERPRETATION
+
+Cake, also known as sch_cake is a modern bandwidth limiter, which eliminates
+buffer bloat over slow links. It's also capable to give flows, hosts and each 
+flow of each host a fair part of the avaible bandwidth.
+
+This plugin allows for a monitor of the pressure on the qdisc, by monitoring key
+values.
+
+=head1 SEE ALSO
+
+Take a look at "man cake" to get more information about cake.
+
+=head1 MAGIC MARKERS
+
+ #%# family=auto
+ #%# capabilities=autoconf suggest
+
+=head1 AUTHORS
+
+RubenKelevra <ruben@vfn-nrw.de>
+
+work based on the tc plugin, authors:
+Steve Schnepp <steve.schnepp@gmail.com>,
+Samuel Smith <esaym@cpan.org>,
+Nye Liu <nyet@nyet.org>
+
+=head1 LICENSE
+
+GPLv2 or later
+
+=cut
+
+DEVICE=${0##*/cake_}
+
+tc_get_ifb_dev() {
+	dev="$(/sbin/tc filter show dev $DEVICE parent ffff: protocol all | grep "mirred" | grep "Egress Redirect to device")"
+	[ $? -eq 1 ] && return 1
+	
+	echo "$dev" | sed -n 's/.* device \([^ ]*\).*/\1/p' | tr ')' ' ' | awk '{ print $1 }'
+}
+
+tc_cake_sent() {
+	/sbin/tc -s qdisc show dev "$1" | grep -E "^ Sent" | tr ',' ' ' | tr ')' ' ' 
+}
+tc_cake_backlog() {
+	/sbin/tc -s qdisc show dev "$1" | grep -E "^ backlog" | tr 'p' ' ' 
+}
+
+DEVICE_IN="$(tc_get_ifb_dev)"
+has_if_in=$?
+
+case "$1" in
+	autoconf)
+	if [ -r /proc/net/dev ]; then
+		echo yes
+		exit 0
+	else
+		echo "no (/proc/net/dev not found)"
+		exit 1
+	fi
+	;;
+	suggest)
+	if [ -r /proc/net/dev ]; then
+		ifs="$(awk '
+				/^ *(eth|tap|bond|wlan|ath|ra|sw|eno|ens|enp|wlp|wl)[0-9]*/ {
+				   split($0, a, /: */);
+				   gsub(/^ +/,"",a[1]);
+				   if (($2 > 0) || ($10 > 0)) print a[1]; }' /proc/net/dev)"
+		cake_ifs=()
+		for if in $ifs; do
+			qdisc="$(/sbin/tc -s qdisc show dev "$if" | head -n1 | awk '{ print $2 }')"
+			if [ "$qdisc" == "cake" ]; then
+				cake_ifs+=("$if")
+			fi
+		done
+		echo "$cake_ifs"
+	fi
+	exit 0
+	;;
+	config)
+	
+	if [ $has_if_in -eq 0 ]; then
```

Comment:
Please quote variables. We are striving for `shellcheck` cleanliness in the plugins.
See the [hints for new plugins](https://github.com/munin-monitoring/contrib/tree/master/plugins#submit-a-new-plugin).

Reply:
shellcheck has 255 MB dependencies. I'm sorry, I won't bloat my system with this tool.

Reply:
Well, shellcheck isn't complaining about this line, what exactly is wrong with it?

integer comparisons AFAIK cannot be double quoted and since it's not an input variable I don't see any point changing this to a string, just to make any sanitizer happy.

Reply:
> Well, shellcheck isn't complaining about this line, what exactly is wrong with it?

Correct: in this case it does not complain, since you are assigning only fixed strings to this variable. Overly simple static parsers (e.g.: humans like me) will repeatedly stumble over this, nevertheless :)

> integer comparisons AFAIK cannot be double quoted

Please rest assured, that the shell parser will *never* forbid you to double-quote a variable :)
(double quotes only prevent word splitting for various expansions - this is usually what you want)

Reply:
Hehe - yes, that works. I am writing way too much shell code for busybox's ash. All the usual features of `dash` are supported there.